### PR TITLE
[hailctl] Add creator label to `hailctl start`

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: hail/python/hailctl/dataproc/start.py [hunk 1]
```diff
@@ -172,6 +173,9 @@ def main(args, pass_through_args):
     if args.bucket:
         conf.flags['bucket'] = args.bucket
 
+    label = safe_call('gcloud', 'config', 'get-value', 'account')
+    conf.flags['labels'] = 'creator=' + label.replace('.', '-dot-').replace('@', '-at-')  # no special chars
```
Comment (cseed) [inline hail/python/hailctl/dataproc/start.py]: This seems insufficient to be reliable.  From: https://cloud.google.com/resource-manager/docs/creating-managing-labels

 > Values can be empty, and have a maximum length of 63 characters.  ... Keys and values can contain only lowercase letters, numeric characters, underscores, and dashes.  

Emails can have other characters (upper case, +, etc.)

Comment (tpoterba) [reply hail/python/hailctl/dataproc/start.py]: yeah, you're totally right. I'll write something that checks against a regex and defaults to not labeling and printing an error.

Comment (cseed) [reply hail/python/hailctl/dataproc/start.py]: That means if the user has a weird email they can't use gcloud?  Seems rude.

I was going to suggest just munging it: lowercase-ify, replace expected characters readably like you did, change other invalid characters to just _, and then truncate.  As long as this preserves enough information that Sam can figure out who ran it, we're golden.

Comment (cseed) [reply hail/python/hailctl/dataproc/start.py]: Err, can't use hailctl.

**Comment by tpoterba** [comment]
Can use hailctl, just won't get the sam-requested billing labels

File: hail/python/hailctl/dataproc/start.py (hunk 1 above)
Comment (tpoterba) [reply hail/python/hailctl/dataproc/start.py]: but yeah, your solution is better

**Comment by cseed** [comment]
> printing an error

Error sounds like you're going to exit and not do anything.  But you mean print out a message and don't label but still create the cluster?  That seems fine, although munging seems more informational, that is, it's an error to us/Sam, not the user, esp. since they can't do anything about it.

**Comment by tpoterba** [comment]
> print out a message and don't label but still create

That's what I meant, yeah. But I'll do the munging.

**Pull request merged**